### ISSUE TITLE
Unclear what the arg parsing is doing here. Seems to eat the first command. 

### ISSUE THREAD

**Issue by unphased** [watchexec/watchexec#835]
```
❯ watchexec -vvv --log-file we_log.log -- bash -c 'echo foo; echo bar;'      
[Running: bash -c echo foo; echo bar;]

bar
[Command was successful]
```

```
❯ watchexec -- "bash -c 'echo foo; echo bar;'"        
[Running: bash -c 'echo foo; echo bar;']
foo
bar
[Command was successful]

❯ watchexec --version         
watchexec 2.1.1 (2024-05-25) +pid1
build-date: 2024-05-25
release: 2.1.1
features: default,pid1
```

`we_log.log`:

[we_log.log](https://github.com/watchexec/watchexec/files/15444250/we_log.log)

`uname -a`:
```
Darwin Stevens-MBP 23.4.0 Darwin Kernel Version 23.4.0: Fri Mar 15 00:10:42 PDT 2024; root:xnu-10063.101.17~1/RELEASE_ARM64_T6000 arm64
```

Further testing on an Ubuntu machine: 

```
❯ watchexec bash -c 'echo foo; echo bar;'                                                                                         []
[Running: bash -c echo foo; echo bar;]

bar
[Command was successful]
[[Error (not fatal)]]
Native fs watcher error
[Running: bash -c echo foo; echo bar;]

bar
[Command was successful]
[Running: bash -c echo foo; echo bar;]

bar
[Command was successful]
[Running: bash -c echo foo; echo bar;]
```

This is even weirder because it seems to hang for a few seconds before it launches with `[[Error (not fatal)]] Native fs watcher error`. The "Native fs watcher error" is off topic surely but just pointing out the arg parse or whatever behavior is identical here on linux.

On both machines installed via `cargo install watchexec-cli`.

**Comment by unphased** [comment]
Notably, `watchexec -- echo foo` and `watchexec -- 'echo foo; echo bar;'` all work as expected with the `foo` visible.

**Comment by passcod** [comment]
Yeah I think this is _technically_ expected behaviour, if a bit obnoxious. From help page:

> Arguments:
>   [COMMAND]...
>
>   Command to run on changes
> 
>   It's run when events pass filters and the debounce period (and once at startup unless '--postpone' is given). If you pass flags to the command, you should separate it with -- though that is not strictly required.
> 
>   Examples:
> 
>   $ watchexec -w src npm run build
> 
>   $ watchexec -w src -- rsync -a src dest
> 
>   Take care when using globs or other shell expansions in the command. Your shell may expand them before ever passing them to Watchexec, and the results may not be what you expect. Compare:
> 
>   $ watchexec echo src/*.rs
> 
>   $ watchexec echo 'src/*.rs'
> 
>   $ watchexec --shell=none echo 'src/*.rs'
> 
>   **Behaviour depends on the value of '--shell': for all except 'none', every part of the command is joined together into one string with a single ascii space character, and given to the shell as described in the help for '--shell'.** For 'none', each distinct element the command is passed as per the execvp(3) convention: first argument is the program, as a path or searched for in the 'PATH' environment variable, rest are arguments.

(emphasis added at the end)

If you're providing your own shell, what you're looking for is the `-n`/`--shell=none` mode. Otherwise, it does a best-effort join-up, which in this case screws up your command.
